kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java

```diff
@@ -150,37 +156,34 @@ public PrivateKey call() {
               Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
             }
             PEMKeyPair keys = (PEMKeyPair) new PEMParser(new InputStreamReader(keyInputStream)).readObject();
-            return new
-              JcaPEMKeyConverter().
-              getKeyPair(keys).
-              getPrivate();
+            return new JcaPEMKeyConverter().getKeyPair(keys).getPrivate();
           } catch (IOException exception) {
             exception.printStackTrace();
           }
           return null;
         }
       }.call();
     } catch (NoClassDefFoundError e) {
-      throw new KubernetesClientException("JcaPEMKeyConverter is provided by BouncyCastle, an optional dependency. To use support for EC Keys you must explicitly add this dependency to classpath.");
+      throw new KubernetesClientException(
+          "JcaPEMKeyConverter is provided by BouncyCastle, an optional dependency. To use support for EC Keys you must explicitly add this dependency to classpath.");
     }
   }
 
-  private static PrivateKey handleOtherKeys(InputStream keyInputStream, String clientKeyAlgo) throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+  private static PrivateKey handleOtherKeys(InputStream keyInputStream, String clientKeyAlgo)
+      throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
     byte[] keyBytes = decodePem(keyInputStream);
-    KeyFactory keyFactory = KeyFactory.getInstance(clientKeyAlgo);
     try {
       // First let's try PKCS8
-      return keyFactory.generatePrivate(new PKCS8EncodedKeySpec(keyBytes));
+      return KeyFactory.getInstance(clientKeyAlgo).generatePrivate(new PKCS8EncodedKeySpec(keyBytes));
     } catch (InvalidKeySpecException e) {
       // Otherwise try PKCS8
       RSAPrivateCrtKeySpec keySpec = PKCS1Util.decodePKCS1(keyBytes);
-      return keyFactory.generatePrivate(keySpec);
+      return KeyFactory.getInstance(clientKeyAlgo).generatePrivate(keySpec);
```

Comment:
The fix is brittle without a test that ensures this won't be reverted in the future.
A test would be good, but at least a comment pointing to the issue should be added here.